src/topology.jl

```diff
@@ -1,62 +1,78 @@
 """
-    Dims_create!(nnodes::Integer, [ndims::Integer], dims)
+    newdims = Dims_create(nnodes::Integer, dims)
 
-Create a division of `nnodes` processes in a Cartesian grid.
+A convenience function for selecting a balanced Cartesian grid of a total of `nnodes`
+nodes, for example to use with [`MPI.Cart_create`](@ref).
+
+`dims` is an array or tuple of integers specifying the number of nodes in each dimension.
+The function returns an array `newdims` of the same length, such that if `newdims[i] =
+dims[i]` if `dims[i]` is non-zero, and `prod(newdims) == nnodes`, and values `newdims` are
+as close to each other as possible.
+
+`nnodes` should be divisible by the product of the non-zero entries of `dims`.
 
 # External links
 $(_doc_external("MPI_Dims_create"))
 """
-function Dims_create!(nnodes::Integer, ndims::Integer, dims::MPIBuffertype{T}) where {T <: Integer}
-    # int MPI_Dims_create(int nnodes, int ndims, int dims[])
+function Dims_create(nnodes::Integer, dims)
+    dims = Cint[dim for dim in dims]
     @mpichk ccall((:MPI_Dims_create, libmpi), Cint,
-                  (Cint, Cint, Ptr{Cint}), nnodes, ndims, dims)
+                  (Cint, Cint, Ptr{Cint}), nnodes, length(dims), dims)
+    return dims
 end
 
-function Dims_create!(nnodes::Integer, dims::AbstractArray{T,N}) where {T<:Integer, N}
-    cdims = Cint.(dims[:])
-    ndims = length(cdims)
-    Dims_create!(nnodes, ndims, cdims)
-    dims[:] .= cdims
-end
 
 """
-    comm_cart = Cart_create(comm_old::Comm, [ndims::Integer], dims, periods, reorder)
+    comm_cart = Cart_create(comm::Comm, dims; periodic=map(_->false, dims), reorder=false)
```

Comment:
Any thoughts on what should be the defaults? Or if we should have them at all?

Reply:
I agree with both `periodic` and `reorder` being optional.

`periodic` being `false` along all dimensions seems like a reasonable default to me. I don't know if it matters, but this argument is called `periods` in the OpenMPI and MPICH documentations. I do prefer `periodic` though.

As for `reorder`, I'd say leaving it as `false` by default also makes sense, as it would avoid any surprises (reordered ranks) to potentials users who wouldn't want to bother with optional arguments.